src/nni_manager/training_service/remote_machine/remoteMachineData.ts

```diff
@@ -83,101 +84,71 @@ export class RemoteMachineTrialJobDetail implements TrialJobDetail {
- */
-export class SSHClient {
-    private readonly sshClient: Client;
-    private usedConnectionNumber: number; //count the connection number of every client
-    constructor(sshClient: Client, usedConnectionNumber: number) {
-        this.sshClient = sshClient;
-        this.usedConnectionNumber = usedConnectionNumber;
-    }
-
-    public get getSSHClientInstance(): Client {
-        return this.sshClient;
-    }
-
-    public get getUsedConnectionNumber(): number {
-        return this.usedConnectionNumber;
-    }
-
-    public addUsedConnectionNumber(): void {
-        this.usedConnectionNumber += 1;
-    }
-
-    public minusUsedConnectionNumber(): void {
-        this.usedConnectionNumber -= 1;
-    }
-}
-
 /**
  * The remote machine ssh client manager
  */
 export class SSHClientManager {
-    private readonly sshClientArray: SSHClient[];
+    private readonly sshExecutorArray: ShellExecutor[];
     private readonly maxTrialNumberPerConnection: number;
     private readonly rmMeta: RemoteMachineMeta;
-    constructor(sshClientArray: SSHClient[], maxTrialNumberPerConnection: number, rmMeta: RemoteMachineMeta) {
+    constructor(sshExecutorArray: ShellExecutor[], maxTrialNumberPerConnection: number, rmMeta: RemoteMachineMeta) {
         this.rmMeta = rmMeta;
-        this.sshClientArray = sshClientArray;
+        this.sshExecutorArray = sshExecutorArray;
         this.maxTrialNumberPerConnection = maxTrialNumberPerConnection;
     }
 
     /**
      * find a available ssh client in ssh array, if no ssh client available, return undefined
      */
-    public async getAvailableSSHClient(): Promise<Client> {
-        const deferred: Deferred<Client> = new Deferred<Client>();
-        for (const index of this.sshClientArray.keys()) {
-            const connectionNumber: number = this.sshClientArray[index].getUsedConnectionNumber;
+    public async getAvailableSshExecutor(): Promise<ShellExecutor> {
+        for (const index of this.sshExecutorArray.keys()) {
+            const connectionNumber: number = this.sshExecutorArray[index].getUsedConnectionNumber;
             if (connectionNumber < this.maxTrialNumberPerConnection) {
-                this.sshClientArray[index].addUsedConnectionNumber();
-                deferred.resolve(this.sshClientArray[index].getSSHClientInstance);
+                this.sshExecutorArray[index].addUsedConnectionNumber();
 
-                return deferred.promise;
+                return this.sshExecutorArray[index];
             }
         }
 
         //init a new ssh client if could not get an available one
-        return this.initNewSSHClient();
+        return await this.initNewSSHClient();
     }
 
     /**
      * add a new ssh client to sshClientArray
      * @param sshClient SSH Client
      */
-    public addNewSSHClient(client: Client): void {
-        this.sshClientArray.push(new SSHClient(client, 1));
+    public addNewSSHClient(executor: ShellExecutor): void {
```

Comment:
In this class, the concept of SSHClient seems be replaced by SSHExecutor. Suggest to unify variable and function names.

Reply:
udpated